token.go

```diff
@@ -218,6 +219,13 @@ func (t *simpleTokenManager) getRetrieveNewTokenClosure(uri *url.URL) func(chan
 		t.token = &token
 		config.Set(configBearerToken, token.AccessToken)
 
+		//don't block on the buffered channel.  that means there is already a signal to serve new token
+		select {
+		case tokenUpdatedApiNotificationChan <- true:
```

Comment:
I would suggest wrap `tokenUpdatedApiNotificationChan ` into a member of TokenManager, instead of using global var.
You can have a method of `simpleTokenManager`:
```
func (s *simpleTokenManager) retrieveApiTokenReady() bool {
  return <-tokenUpdatedApiNotificationChan
}
```
In api.go you can use `apidTokenManager.retrieveApiTokenReady()`.
This provides better style, better isolation between tests, and make it more unit-testable, especially considering in the future there may be something like api_test.go.